src/test/java/teammates/test/cases/webapi/GetFeedbackQuestionRecipientsActionTest.java

```diff
@@ -23,12 +52,190 @@ protected String getRequestMethod() {
+                session.getCourseId(), questionNumber);
+        return new String[] {
+                Const.ParamsNames.FEEDBACK_QUESTION_ID, question.getFeedbackQuestionId(),
+                Const.ParamsNames.INTENT, intent.toString(),
+                Const.ParamsNames.FEEDBACK_SESSION_MODERATED_PERSON, moderatedPerson,
+                Const.ParamsNames.PREVIEWAS, previewPerson,
+                Const.ParamsNames.REGKEY, regKey,
+        };
+    }
+
+    private FeedbackQuestionRecipientsData getRecipients(String[] params) {
+        verifyCanAccess(params); // Make sure not violates access control
+        GetFeedbackQuestionRecipientsAction action = getAction(params);
+        JsonResult result = getJsonResult(action);
+        assertEquals(HttpStatus.SC_OK, result.getStatusCode());
+        return (FeedbackQuestionRecipientsData) result.getOutput();
+    }
+
+    private void verifyFeedbackQuestionRecipientsDataEquals(FeedbackQuestionRecipientsData expected,
+                                                            FeedbackQuestionRecipientsData actual) {
+        assertEquals(expected.getRecipients().size(), actual.getRecipients().size());
+        for (int i = 0; i < expected.getRecipients().size(); i++) {
+            assertEquals(expected.getRecipients().get(i).getIdentifier(),
+                    actual.getRecipients().get(i).getIdentifier());
+            assertEquals(expected.getRecipients().get(i).getName(), actual.getRecipients().get(i).getName());
+        }
+    }
+
+    private void testModeratedAndPreviewPersonToGetRecipients() {
+
+        loginAsStudent(student1InCourse1.googleId);
+        // A copy of recipients got from giver without moderate or preview person.
+        String[] questionParams = generateParameters(firstSessionInCourse1, 2,
+                Intent.STUDENT_SUBMISSION, "", "", "");
+        FeedbackQuestionRecipientsData originRecipients = getRecipients(questionParams);
+
+        ______TS("Test moderated person get recipient, should be same as recipients got from giver");
+        loginAsInstructor(instructor1OfCourse1.googleId);
+        String[] moderateQuestionParams = generateParameters(firstSessionInCourse1, 2,
+                Intent.STUDENT_SUBMISSION, "", student1InCourse1.email, "");
+        FeedbackQuestionRecipientsData moderatedRecipients = getRecipients(moderateQuestionParams);
+        verifyFeedbackQuestionRecipientsDataEquals(originRecipients, moderatedRecipients);
+
+        ______TS("Test preview person get recipient, should be same as recipients got from giver");
+        String[] previewQuestionParams = generateParameters(firstSessionInCourse1, 2,
+                Intent.STUDENT_SUBMISSION, "", "", student1InCourse1.email);
+        FeedbackQuestionRecipientsData previewRecipients = getRecipients(previewQuestionParams);
+        verifyFeedbackQuestionRecipientsDataEquals(originRecipients, previewRecipients);
+    }
+
+    private void testDifferentRecipientTypes() {
```

Comment:
Same here, add a @Test

Reply:
👍 Also the test case naming convention? 
`testExecute_differentRecipientTypes_shouldXXX`?